pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/TopicLookupBase.java

```diff
@@ -333,13 +334,16 @@ public static CompletableFuture<ByteBuf> lookupTopicAsync(PulsarService pulsarSe
 
     private static void handleLookupError(CompletableFuture<ByteBuf> lookupFuture, String topicName, String clientAppId,
                                    long requestId, Throwable ex){
-        final Throwable unwrapEx = FutureUtil.unwrapCompletionException(ex);
+        Throwable unwrapEx = FutureUtil.unwrapCompletionException(ex);
         final String errorMsg = unwrapEx.getMessage();
+        if (unwrapEx instanceof PulsarServerException) {
+            unwrapEx = FutureUtil.unwrapCompletionException(unwrapEx.getCause());
```

Comment:
Are you sure `unwrapEx.getCause()` must have a not-null value? :)

Reply:
```java
        final var ex = new PulsarServerException("hello");
        final var unwrapEx = FutureUtil.unwrapCompletionException(ex.getCause());
        System.out.println(unwrapEx == null);
        System.out.println(unwrapEx instanceof IllegalStateException);
        System.out.println(unwrapEx instanceof MetadataStoreException);
```

outputs:

```
true
false
false
```

When `unwrapEx.getCause()` is null, it will reach the `else` block that `ServiceNotReady` is returned to the client.